app/controllers/api/base_controller.rb

```diff
@@ -52,6 +52,13 @@ class BaseController < ActionController::Base
 
     rescue_from Foreman::MaintenanceException, :with => :service_unavailable
 
+    rescue_from ActionDispatch::Http::Parameters::ParseError do |exception|
+      error_output = "There was a problem in the JSON you submitted: #{exception.cause.message}"
+      Rails.logger.debug(error_output)
+
+      render status: :bad_request, json: { status: 400, error: error_output }
```

Comment:
My only concern with this particular change is that it previously was only done if the content was JSON or the client explicitly said it accepted JSON. Though I'm not sure how much of an issue it is in practice. Do we have any YAML or XML endpoints?

Then again, it will effectively be changed from a HTTP 500 (which I'm guessing is HTML output) to HTTP 400 with JSON output. Probably not too bad.

Reply:
I don't think we do anything but JSON in the API, no. It was accepting and returning JSON only already:

https://github.com/theforeman/foreman/blob/85ef2e4b41c12bb32420678b5035246f2cffcfca/app/controllers/api/base_controller.rb#L15
https://github.com/theforeman/foreman/blob/85ef2e4b41c12bb32420678b5035246f2cffcfca/app/controllers/api/base_controller.rb#L179-L182

Reply:
Ah, I missed that this is the API controller. Then I do wonder if there's non-API endpoints which accept JSON. Probably not

Reply:
None that would be tested ;-)

Reply:
And then they should just send well formatted JSON ;)